app/src/main/java/blockeq/com/stellarwallet/activities/SendActivity.kt

```diff
@@ -1,48 +1,77 @@
 package blockeq.com.stellarwallet.activities
 
+import android.app.Activity
+import android.content.Intent
 import android.os.Bundle
-import android.widget.TextView
+import android.view.View
 import android.widget.Toast
 import blockeq.com.stellarwallet.R
+import blockeq.com.stellarwallet.WalletApplication
+import blockeq.com.stellarwallet.interfaces.OnSendPayment
+import blockeq.com.stellarwallet.models.PinType
+import blockeq.com.stellarwallet.services.networking.Horizon
+import blockeq.com.stellarwallet.services.networking.Horizon.Companion.getBalance
+import blockeq.com.stellarwallet.utils.NetworkUtils
+import blockeq.com.stellarwallet.utils.StringFormat.Companion.getNumDecimals
+import blockeq.com.stellarwallet.utils.StringFormat.Companion.hasDecimalPoint
 import com.davidmiguel.numberkeyboard.NumberKeyboardListener
-import kotlinx.android.synthetic.main.activity_base_popup.*
 import kotlinx.android.synthetic.main.contents_send.*
 
 
-class SendActivity : BasePopupActivity(), NumberKeyboardListener {
-
-    private val MAX_ALLOWED_DECIMALS = 4
-    val ADDRESS_DATA = "ADDRESS"
+class SendActivity : BasePopupActivity(), NumberKeyboardListener, OnSendPayment {
 
+    companion object {
+        const val MAX_ALLOWED_DECIMALS = 4
+        const val ADDRESS_DATA = "ADDRESS"
+    }
 
     private var amountText: String = ""
     private var amount: Double = 0.0
-
-    var availableBalance = "6.02 XLM"
-
-    override fun setTitle(): Int {
-        return R.string.title_activity_my_wallet
-    }
+    private var address: String = ""
 
     override fun setContent(): Int {
         return R.layout.contents_send
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        titleText.text = availableBalance
+        titleText.text = "Available: " + getBalance() + " XLM"
 
-        amount_text.text = "0"
+        amountTextView.text = "0"
         numberKeyboard.setListener(this)
 
-        val address = intent.getStringExtra(ADDRESS_DATA)
+        address = intent.getStringExtra(ADDRESS_DATA)
+        addressEditText.text = address
 
-        findViewById<TextView>(R.id.address_text).apply {
-            text = address
-        }
+        send_button.setOnClickListener {
+            val data = WalletApplication.localStore!!.encryptedPhrase
 
+            if (data != null && !data.isEmpty()) {
+                launchPINView(PinType.CHECK, "", data, false)
+            }
+        }
+    }
 
-        send_button.setOnClickListener { Toast.makeText(this, "send clicked", Toast.LENGTH_LONG).show() }
+    //region User Interface
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == PinActivity.PIN_REQUEST_CODE) {
+            when (resultCode) {
+                Activity.RESULT_OK -> {
+                    if (NetworkUtils(this).isNetworkAvailable()) {
```

Comment:
We can put a TODO here for dealing with network states - it would be great to move them wit the network calls itself.